### ISSUE TITLE
[@typescript-eslint/indent] Wrong enum error

### ISSUE THREAD

**Issue by FStefanni** [typescript-eslint/typescript-eslint#3357]
- [x ] I have tried restarting my IDE and the issue persists.
- [x ] I have updated to the latest version of the packages.
- [x ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
   "@typescript-eslint/indent": [
                    "error", 4,
                    {
                        "CallExpression": {
                            "arguments": "first"
                        },
                        "FunctionDeclaration": {
                            "parameters": 1,
                            "body": 1
                        },
                        "SwitchCase": 1
                    }
                ]
  }
}
```

```TS
export enum MyEnum
{
    MY_VALUE_01
}
```

**Expected Result**

No error/warning

**Actual Result**

```
Expected indentation of 4 spaces but found 0. eslint(@typescript-eslint/indent)
```

**Additional Info**

* No exception
* If the opening curl is on the same line of the enum declaration, the error disappears

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.1` |
| `@typescript-eslint/parser`        | `4.22.1` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.25.0` |
| `node`                             | `15.7.0` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/1824

Please use the search or look at the pinned issues before raising an issue.
